### ISSUE TITLE
TS service crashes repeatedly

### ISSUE THREAD

**Issue by Mobius0526** [vuejs/language-tools#4480]
Type: <b>Bug</b>

I found that the TS server often restarts,It's more likely to occur when I switch between different .vue files

Extension version: 2.0.21
VS Code version: Code 1.90.1 (611f9bfce64f25108829dd295f54a6894e87339d, 2024-06-11T21:01:24.262Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Remote OS version: Linux x64 5.15.146.1-microsoft-standard-WSL2
Remote OS version: Linux x64 5.15.146.1-microsoft-standard-WSL2

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz (4 x 2712)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|15.94GB (2.93GB free)|
|Process Argv|--folder-uri vscode-remote://wsl%2Bdebian/home/mobius/work/shihuan-multimedia-ui --crash-reporter-id dc2a9fc6-7d55-4fa3-9be5-45db5215ed52|
|Screen Reader|no|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Debian|
|OS|Linux x64 5.15.146.1-microsoft-standard-WSL2|
|CPUs|Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz (4 x 0)|
|Memory (System)|5.79GB (0.45GB free)|
|VM|0%|

|Item|Value|
|---|---|
|Remote|WSL: Debian|
|OS|Linux x64 5.15.146.1-microsoft-standard-WSL2|
|CPUs|Intel(R) Core(TM) i5-6400 CPU @ 2.70GHz (4 x 0)|
|Memory (System)|5.79GB (0.45GB free)|
|VM|0%|
</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscorecescf:30445987
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
pythongtdpath:30769146
pythonidxpt:30866567
pythonnoceb:30805159
asynctok:30898717
pythontestfixt:30902429
pythonregdiag2:30936856
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
dsvsc016:30899300
dsvsc017:30899301
dsvsc018:30899302
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
jchc7451:31067544
dsvsc021:30996838
bdiig495:31013172
pythoncenvpt:31062603
a69g1124:31058053
dvdeprecation:31068756
pythonprt:31056678
dwnewjupyter:31046869
legacy_priority:31077528

```

</details>

**Comment by Mobius0526** [comment]
2024-06-19 10:09:56.036 [info] Starting TS Server
2024-06-19 10:09:56.036 [info] Using tsserver from: /home/mobius/work/shihuan-multimedia-ui/node_modules/typescript/lib/tsserver.js
2024-06-19 10:09:56.036 [info] <syntax> Forking...
2024-06-19 10:09:56.036 [info] <syntax> Starting...
2024-06-19 10:09:56.036 [info] <semantic> Forking...
2024-06-19 10:09:56.036 [info] <semantic> Starting...
2024-06-19 13:37:06.063 [error] TSServer exited. Code: null. Signal: SIGABRT
2024-06-19 13:37:06.069 [info] Starting TS Server
2024-06-19 13:37:06.069 [info] Using tsserver from: /home/mobius/work/shihuan-multimedia-ui/node_modules/typescript/lib/tsserver.js
2024-06-19 13:37:06.069 [info] <syntax> Forking...
2024-06-19 13:37:06.069 [info] <syntax> Starting...
2024-06-19 13:37:06.069 [info] <semantic> Forking...
2024-06-19 13:37:06.069 [info] <semantic> Starting...
2024-06-19 15:36:44.489 [error] TSServer exited. Code: null. Signal: SIGABRT
2024-06-19 15:36:44.503 [info] Starting TS Server
2024-06-19 15:36:44.503 [info] Using tsserver from: /home/mobius/work/shihuan-multimedia-ui/node_modules/typescript/lib/tsserver.js
2024-06-19 15:36:44.503 [info] <syntax> Forking...
2024-06-19 15:36:44.503 [info] <syntax> Starting...
2024-06-19 15:36:44.503 [info] <semantic> Forking...
2024-06-19 15:36:44.503 [info] <semantic> Starting...

**Comment by Mobius0526** [comment]
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Error: read ECONNRESET
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout

**Comment by Mobius0526** [comment]
Vscode prompts many uncaught errors: Cannot read properties of undefined (reading 'match')

**Comment by so1ve** [comment]
@Mobius0526 `match undefined`: https://github.com/vuejs/language-tools/issues/4457#issuecomment-2163968359

**Comment by jshrssll** [comment]
Just came across this issue myself. I found that downgrading from `v2.0.21` to `v2.0.19` solved it for me. Also ensure that you are using the workspace Typescript version in VS Code if applicable to your project. You can change that setting by clicking on the `{}` icon on the bottom right of the window

**Comment by yyrjlw** [comment]
This problem still occurs on version 2.0.26

**Comment by OguzcanKarakoc** [comment]
I keep encountering the same repeated error:

```
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
[Vue Named Pipe Client] Timeout
...
```

This issue is incredibly frustrating and makes it very difficult to program effectively. I'd like to debug this myself, but I don't know where to start. I checked the developer tools console in VSCode for any errors, but found nothing.

It takes an extraordinarily long time for the correct types to appear. Any guidance on how to debug and resolve this issue would be greatly appreciated.

Some extra info:

So I have disabled every extension except **Vue - Official**

I am working in a **nuxt** project with **shadcn-vue**. Essentially following these steps: https://www.shadcn-vue.com/docs/installation/nuxt.html 

Typescript version: **5.5.3**

Vs code version: Version: 1.91.1 (user setup) Commit: f1e16e1e6214d7c44d078b1f0607b2388f29d729 Date: 2024-07-09T22:06:49.809Z Electron: 29.4.0 ElectronBuildId: 9728852 Chromium: 122.0.6261.156 Node.js: 20.9.0 V8: 12.2.281.27-electron.0 OS: Windows_NT x64 10.0.22631

Vue version: **^3.4.31**

  System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 AMD Ryzen 7 2700X Eight-Core Processor
    Memory: 5.86 GB / 15.95 GB
  Binaries:
    Node: 18.18.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 4.3.1 - C:\Program Files\nodejs\yarn.CMD
    npm: 10.2.5 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Chrome: 126.0.6478.127
    Edge: Chromium (126.0.2592.102)
    Internet Explorer: 11.0.22621.3527
